crypto/objects/obj_dat.c

```diff
@@ -68,33 +71,39 @@ DEFINE_RUN_ONCE_STATIC(obj_lock_initialise)
         return 0;
     }
 #endif
+    added = lh_ADDED_OBJ_new(added_obj_hash, added_obj_cmp);
+    if (added == NULL) {
+        objs_free_locks();
+        return 0;
+    }
+
     return 1;
 }
 
-static ossl_inline int ossl_init_added_lock(void)
+static ossl_inline int ossl_init_added_api(void)
 {
 #ifndef OPENSSL_NO_AUTOLOAD_CONFIG
     /* Make sure we've loaded config before checking for any "added" objects */
     OPENSSL_init_crypto(OPENSSL_INIT_LOAD_CONFIG, NULL);
 #endif
-    return RUN_ONCE(&ossl_obj_lock_init, obj_lock_initialise);
+    return RUN_ONCE(&ossl_obj_api_init, obj_api_initialise);
 }
 
 static ossl_inline int ossl_obj_write_lock(int lock)
 {
+    if (!ossl_init_added_api())
```

Comment:
We do not seem to call obj_write_lock() and obj_read_lock() with `lock == 0` argument. Could you please refactor the functions to drop the argument?